pkg/controller/spec_processor_test.go

```diff
@@ -14,45 +14,75 @@ import (
 
 func TestSpecProcessor(t *testing.T) {
 	t.Parallel()
-	sp := SpecProcessor{
-		selfName:  "abc",
-		resources: processedResources(),
-		allowedResources: map[smith_v1.ResourceName]struct{}{
-			"res1": {},
+	sp, err := NewSpec(processedResources(), []smith_v1.Reference{
+		{
+			// Nameless references cause dependencies only.
+			Resource: "resX",
 		},
-	}
+		{
+			Name:     "res1aslice",
+			Resource: "res1",
+			Path:     "a.slice[?(@.label==\"label2\")].value",
```

Comment:
So we are not even referring to `name` inside the path or anywhere else, right? What's the purpose of `name` then?